storage/postgresql/save.go

```diff
@@ -26,12 +26,11 @@ func (b *PostgresBackend) SaveEvent(evt *nostr.Event) error {
         VALUES ($1, $2, $3, $4, $5, $6, $7)
     `, evt.ID, evt.PubKey, evt.CreatedAt.Unix(), evt.Kind, tagsj, evt.Content, evt.Sig)
 	if err != nil {
-		if strings.Index(err.Error(), "UNIQUE") != -1 {
-			// already exists
-			return nil
+		if strings.Contains(err.Error(), "duplicate key value violates unique constraint") {
```

Comment:
searching substrings in error messages is so fragile. i know you didn't add it but consider something like this instead:

```go
res, err := b.DB.Exec("INSERT INTO event (...) VALUES () ON CONFLICT (id) DO NOTHING", ...)
if err != nil {
  return err
}
if res.RowsAffected() == 0 {
  return relayer.ErrDupEvent
}
return nil
```

i haven't tried it, though. just an idea. not sure it works. maybe i'll do it later and send in a separate pull request.

Reply:
Yes, much nicer. I was surprised sqlx lib doesn't already return custom errors for integrity related errors 

 I've tested your solution a bit (tried will inserting `id=NULL` to see if it would get silenced - it doesn't) and after seeing it's working as expected, included it in this PR.